todo_react/todo-react/src/App.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {useState} from 'react';
+import './App.css';
+import {AddToDo} from "./components/AddToDo";
+import {ToDoItem} from "./components/ToDoItem";
+import {ITodo} from "./models";
+import {TodosContext} from "./context"
+
+export const App: React.FC = () => {
+    let todoList = [...JSON.parse(localStorage.getItem('TodoList') || "")]
```

Comment:
1. why need `...` ?
2. can try to clean local storage, fix JSON.parse error
3. avoid parsing data each render (read about react lifecycle)
4. Move state management to a new component
5. Re-render todos list when it's changed in local storage in a different tab

Reply:
some hints
1. `JSON.parse("[]")` returns array. you don't need `...`
2. wrap parsing in `try/catch`
3. you are making `JSON.parse` every time component re-renders, but you need it only once – initial state set (can be done on mounted)
4. move state logic (`useState` and `JSON.parse` to `<TodosProvider>`)
5. it is additional task (try something like `window.addEventListener("storage", () => {})`)